fatjdbc/src/main/java/com/qubole/quark/fatjdbc/executor/DDLPlanExecutor.java

```diff
@@ -98,27 +98,27 @@ public QuarkMetaResultSet execute(ParserResult result)
     }
     if (sqlNode instanceof SqlCreateQuarkDataSource) {
       int id = executeCreateDataSource((SqlCreateQuarkDataSource) sqlNode);
-      connection.resetSelectParser();
+      connection.setIsDirty();
```

Comment:
As discussed offline, de-coupling of DDLPlanExecutor and Connection will be done in separate PR.